app/lib/proxy_config_affecting_changes.rb

```diff
@@ -35,6 +38,33 @@ def issue_proxy_affecting_change_events
         return if changes_attributes.include?('created_at') || (changes_attributes & PROXY_CONFIG_AFFECTING_ATTRIBUTES).empty?
         issue_proxy_affecting_change_event(self)
       end
+
+      def find_or_create_proxy_config_affecting_change
+        proxy_config_affecting_change || create_proxy_config_affecting_change
+      end
+      alias affecting_change_history find_or_create_proxy_config_affecting_change
+      private :find_or_create_proxy_config_affecting_change
+
+      def pending_affecting_changes?
+        return unless apicast_configuration_driven?
+        config = proxy_configs.sandbox.newest_first.first
+        return false unless config
+        config.created_at < affecting_change_history.updated_at
+      end
+    end
+  end
+
+  module BackendApiConfigExtension
+    extend ActiveSupport::Concern
+
+    included do
+      include ProxyConfigAffectingChanges
+
+      after_commit :issue_proxy_affecting_change_events
+
+      def issue_proxy_affecting_change_events
+        backend_api.proxies.each(&method(:issue_proxy_affecting_change_event))
```

Comment:
```suggestion
        backend_api.proxies.find_each(&method(:issue_proxy_affecting_change_event))
```
So if there are many, I feel a bit safer this way than loading them all to memory at once :thinking: